lib/requirements/file_attachment_metadata_checker.rb

```diff
@@ -0,0 +1,30 @@
+module Requirements
+  class FileAttachmentMetadataChecker
+    UNIQUE_REF_MAX_LENGTH = 255
+
+    attr_reader :unique_reference
+
+    def initialize(unique_reference: nil)
```

Comment:
We usually pass the revision and/or all the params to the checker, and let it decide what data it needs. This makes it easier to add more checks, since we don't need to alter any parameters. The [image checker](https://github.com/alphagov/content-publisher/blob/master/lib/requirements/image_revision_checker.rb) is a good example of how we've done it for something similar, could we follow that approach here?

Reply:
I'm inclined to leave this for now, as it's not necessary to change it here, and the need for it to change will become clear in future work. For now, it's consistent with the checker for file attachments ([in future](https://trello.com/c/3E2vqBkS/1507-support-official-documents-for-featured-file-attachments), we'll be renaming this to `FileAttachmentUploadChecker`, as part of supporting issues for file attachments on the document summary).